### ISSUE TITLE
Enable snapping to the center of a shape while editing and snap to a center of an edge while editing.

### ISSUE THREAD

**Issue by atorkmabrains** [KLayout/klayout#1335]
Currently, snapping doesn't support snapping to center. It only supports corners and general locations on edges. But It would be very useful to allow the snapping the center of edges and objects while editing.

**Comment by Partmedia** [comment]
I agree. The path tool already has something like you describe to continue existing paths, but this would also be useful for any shape.
![image](https://github.com/KLayout/klayout/assets/3229565/b89cf6a4-d387-4d78-b5cc-97910b86e502)
Aside: I would also find a "continue path width" setting that sets the width of the new path to be the width of the old one.

**Comment by atorkmabrains** [comment]
@Partmedia Yes, we used path as possible alternative. But still haven't such capability on polygons and edges in general is extremely helpful for manual layout.
